### PR TITLE
[FLINK-13873][metrics] Change the column family as tags for influxdb …

### PR DESCRIPTION
**Pull request by Mrart**
### **What is the purpose of the change**

When we use influxdb reporter report rocksdb metric data，influxdb reporter call {#getLogicalScope()} get the measurements,and call {#getAllVariables()} get tags. and we need column family as tags for group by in influxSQL. In this PR,  change addgroup(columnFamilyName) to addgroup("column_family", columnFamilyName) build metric group, This will help build column family as tags in influx reporter

### **Brief change log**

 change addgroup(columnFamilyName) to addgroup("column_family", columnFamilyName) build metric group

### **Verifying this change**

This change is a trivial rework / code cleanup without any test coverage

### **Does this pull request potentially affect one of the following parts:**

*  Dependencies (does it add or upgrade a dependency): (no)
* The public API, i.e., is any changed class annotated with @Public(Evolving): (no.)
* The serializers: (no)
* The runtime per-record code paths (performance sensitive): (no)
* Anything that affects deployment or recovery: JobManager (and its components),  Checkpointing, Yarn/Mesos, ZooKeeper: (no)
* The S3 file system connector: (no)

### **Documentation**

Does this pull request introduce a new feature? yes
If yes, how is the feature documented?  javadocs

### CODE REVIEW

**Comment by Mrart** [comment]
> This change breaks existing setups. We may want to introduce a config option to control whether column_family should be exposed as a variable or not.
Add a config option maybe too heaviest for this ?

**Comment by Mrart** [comment]
Add a config option maybe too heaviest for this ?

**Comment by zentol** [comment]
I wouldn't usually recommend adding options for things like this, but as I said this can break user setups since the MetricIdentifier for these metrics now contains an additional constant.

**Comment by Mrart** [comment]
@zentol  It seems a mistake, I dont go CI again。I dont know why?

**Comment by Mrart** [comment]
@zentol  I have test this code，But I dont know how to run CI again？It dont run auto。

**Comment by Mrart** [comment]
@rmetzger  It seems  a ci failed,  I have not seem any log about the failed. Cloud you help me?

**Comment by Mrart** [comment]
@zentol  Could you help review again，I have add  A config option。I sorry for delay。

**Comment by Mrart** [comment]
> This change breaks existing setups. We may want to introduce a config option to control whether column_family should be exposed as a variable or not.
 I have add a config option like "state.backend.rocksdb.column-family-as-variable" for control the column_family

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackendBuilder.java [hunk 1]
```diff
@@ -218,6 +222,11 @@ public RocksDBKeyedStateBackendBuilder(
 		return this;
 	}
 
+	RocksDBKeyedStateBackendBuilder<K> setColumnFamilyAsVariable(boolean isVariable) {
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackendBuilder.java]: We can make this part of the `RocksDBNativeMetricOptions` which should simplify the PR quite a bit.

Comment (Mrart) [reply flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackendBuilder.java]: I get your idea, I will edit again.

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java [hunk 2]
```diff
@@ -149,6 +151,13 @@
 		.key(RocksDBProperty.ActualDelayedWriteRate.getConfigKey())
 		.defaultValue(false)
 		.withDescription("Monitor the current actual delayed write rate. 0 means no delay.");
+
+	public static final ConfigOption<Boolean> COLUMN_FAMILY_AS_VARIABLE = ConfigOptions
+		.key(RocksDBProperty.ColumnFamilyAsVariable.getConfigKey())
+		.defaultValue(false)
+		.withDescription(String.format("The column family as variable for RocksDB metrics to use column family as variable or not, when create metric group. " +
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java]:
```suggestion
		.withDescription(String.format("Whether to expose the column family as a variable." +
```

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricMonitor.java [hunk 3]
```diff
@@ -71,7 +73,9 @@ public RocksDBNativeMetricMonitor(
 	 * @param handle native handle to the column family
 	 */
 	void registerColumnFamily(String columnFamilyName, ColumnFamilyHandle handle) {
-		MetricGroup group = metricGroup.addGroup(columnFamilyName);
+
+		Boolean columnFamilyAsVariable = options.isEnableColumnFaminlyAsVariable();
+		MetricGroup group = columnFamilyAsVariable ?  metricGroup.addGroup(COLUMN_FAMILY_KEY, columnFamilyName) : metricGroup.addGroup(columnFamilyName);
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricMonitor.java]:
```suggestion
		MetricGroup group = columnFamilyAsVariable
			? metricGroup.addGroup(COLUMN_FAMILY_KEY, columnFamilyName)
			: metricGroup.addGroup(columnFamilyName);
```

File: flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptionsTest.java [hunk 4]
```diff
@@ -32,8 +32,13 @@ public void testNativeMetricsConfigurable() {
 		for (RocksDBProperty property : RocksDBProperty.values()) {
 			Configuration config = new Configuration();
 			config.setBoolean(property.getConfigKey(), true);
-
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptionsTest.java]: revert

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java [hunk 5]
```diff
@@ -401,11 +414,18 @@ public void enableActualDelayedWriteRate() {
 		this.properties.add(RocksDBProperty.ActualDelayedWriteRate.getRocksDBProperty());
 	}
 
+	/**
+	 * Returns the column family as variable.
+	 */
+	public void enableColumnFamilyAsVariable() {
+		this.properties.add(RocksDBProperty.ColumnFamilyAsVariable.getRocksDBProperty());
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java]: I would add this as a separate flag with a dedicated getter. That way you don't have to modify the assertions in the tests.

**Comment by Mrart** [comment]
@flinkbot run travis

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricMonitor.java [hunk 6]
```diff
@@ -71,7 +73,11 @@ public RocksDBNativeMetricMonitor(
 	 * @param handle native handle to the column family
 	 */
 	void registerColumnFamily(String columnFamilyName, ColumnFamilyHandle handle) {
-		MetricGroup group = metricGroup.addGroup(columnFamilyName);
+
+		boolean columnFamilyAsVariable = options.isColumnFaminlyAsVariable();
+		MetricGroup group = columnFamilyAsVariable
+			?  metricGroup.addGroup(COLUMN_FAMILY_KEY, columnFamilyName)
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricMonitor.java]:
```suggestion
			? metricGroup.addGroup(COLUMN_FAMILY_KEY, columnFamilyName)
```

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java [hunk 7]
```diff
@@ -149,6 +152,13 @@
 		.key(RocksDBProperty.ActualDelayedWriteRate.getConfigKey())
 		.defaultValue(false)
 		.withDescription("Monitor the current actual delayed write rate. 0 means no delay.");
+
+	public static final ConfigOption<Boolean> COLUMN_FAMILY_AS_VARIABLE = ConfigOptions
+		.key(METRICS_COLUMN_FAMILY_AS_VARIABLE_KEY)
+		.defaultValue(false)
+		.withDescription(String.format("Whether to expose the column family as a variable. " +
+			"The column-family-as-variable has default as '%s'", false));
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java]: we don't need the default in the description; the tables in the documentation automatically list them.

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java [hunk 8]
```diff
@@ -239,10 +249,13 @@ public static RocksDBNativeMetricOptions fromConfig(Configuration config) {
 			options.enableActualDelayedWriteRate();
 		}
 
+		options.setColumnFamilyAsVariable(config.getBoolean(COLUMN_FAMILY_AS_VARIABLE));
+
 		return options;
 	}
 
 	private Set<String> properties;
+	private boolean columnFamilyAsVariable;
```
Comment (zentol) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java]: Let's set this to `COLUMN_FAMILY_AS_VARIABLE.defaultValue()` to safeguard against potential future changes.

**Comment by zentol** [comment]
Nearly there ;)

**Comment by Mrart** [comment]
> Nearly there ;)

thks

**Pull request merged**